src/overlays/actors/ovl_En_Geg/z_en_geg.c

```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
+    s16 sp30;
+    f32 sp2C;
+    f32 sp28;
+
+    sp40.x = (Math_SinS(sp32) * 50.0f) + player->actor.world.pos.x;
+    sp40.y = player->actor.world.pos.y;
+    sp40.z = (Math_CosS(sp32) * 50.0f) + player->actor.world.pos.z;
+
+    sp2C = Math_Vec3f_DistXZ(&this->actor.world.pos, &sp40);
+    sp30 = player->actor.world.rot.y - 0x4000;
+
+    sp34.x = (Math_SinS(sp30) * 50.0f) + player->actor.world.pos.x;
+    sp34.y = player->actor.world.pos.y;
+    sp34.z = (Math_CosS(sp30) * 50.0f) + player->actor.world.pos.z;
+
+    if (Math_Vec3f_DistXZ(&this->actor.world.pos, &sp34) < sp2C) {
+        // clang-format off
+        *arg0 = sp40; return arg0;
+        // clang-format on
+    } else {
+        *arg0 = sp34;
+    }
+    return arg0;
+}
+
+u8 func_80BB1B14(EnGeg* this, GlobalContext* globalCtx) {
+    Actor* explosive;
+    Actor* mm = func_ActorCategoryIterateById(globalCtx, NULL, ACTORCAT_ITEMACTION, ACTOR_EN_MM);
+
+    if (mm != NULL) {
+        this->unk_4B0 = Math_Vec3f_Yaw(&this->actor.world.pos, &mm->world.pos);
+        if (func_80BB18FC(this, mm)) {
+            return 1;
+        }
+    }
+
+    explosive = globalCtx->actorCtx.actorList[ACTORCAT_EXPLOSIVES].first;
+
+    while (explosive != NULL) {
+        if ((explosive->id == ACTOR_EN_BOM) && func_80BB18FC(this, explosive)) {
+            this->unk_4B0 = Math_Vec3f_Yaw(&this->actor.world.pos, &explosive->world.pos);
+            if (this->unk_230 & 0x200) {
+                return 0;
+            }
+
+            if (((EnBom*)explosive)->unk_1F9 == 0) {
+                return 2;
+            }
+
+            return 3;
+        }
+        explosive = explosive->next;
+    }
+
+    this->unk_230 &= ~0x200;
+    return 0;
+}
+
+void func_80BB1C1C(EnGeg* this) {
+    u16 i;
+
+    this->unk_49C[0] = this->actor.cutscene;
+
+    for (i = 1; i < ARRAY_COUNT(this->unk_49C); i++) {
+        this->unk_49C[i] = ActorCutscene_GetAdditionalCutscene(this->unk_49C[i - 1]);
+    }
+}
+
+void func_80BB1C8C(EnGeg* this) {
+    if (DECR(this->unk_240) == 0) {
+        this->unk_23E++;
+        if (this->unk_23E >= 3) {
+            this->unk_240 = Rand_S16Offset(30, 30);
+            this->unk_23E = 0;
+        }
+    }
+}
+
+void func_80BB1D04(EnGeg* this) {
+    f32 temp;
+
+    if (this->unk_230 & 1) {
+        temp = this->actor.shape.yOffset;
+    } else {
+        temp = 58.0f;
+    }
+
+    this->actor.focus.pos.x = this->actor.world.pos.x;
+    this->actor.focus.pos.y = temp + this->actor.world.pos.y;
+    this->actor.focus.pos.z = this->actor.world.pos.z;
+    this->actor.focus.rot.x = this->actor.world.rot.x;
+    this->actor.focus.rot.y = this->actor.world.rot.y;
+    this->actor.focus.rot.z = this->actor.world.rot.z;
+}
+
+s32 func_80BB1D64(EnGeg* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    Vec3f sp40;
+    Vec3f sp34;
+    s16 y = (this->actor.yawTowardsPlayer - this->unk_46E) - this->actor.shape.rot.y;
```

Comment:
`y` along usually means y-position while this is dealing with angles and yaw. So you can rename this temp `yaw`